specification/sql/resource-manager/readme.java.md

```diff
@@ -0,0 +1,123 @@
+Please also specify `--azure-libraries-for-java=<path to the root directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(tag) == 'package-pure-2015-05-preview' && $(java) && $(multiapi)
+java:
+  namespace: com.microsoft.azure.management.sql.v2015_05_01_preview
+  output-folder: $(azure-libraries-for-java-folder)/sql/resource-manager/v2015_05_01_preview
+regenerate-manager: true
+generate-interface: true
+```
+
+### Tag: package-pure-fixed-2017-03-previewand java
+
+These settings apply only when `--tag=package-pure-fixed-2017-03-preview --java` is specified on the command line.
+Please also specify `--azure-libraries-for-java=<path to the root directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(tag) == 'package-pure-fixed-2017-03-preview' && $(java) && $(multiapi)
+java:
+  namespace: com.microsoft.azure.management.sql.v2017_03_01_preview
+  output-folder: $(azure-libraries-for-java-folder)/sql/resource-manager/v2017_03_01_preview
+regenerate-manager: true
+generate-interface: true
+```
+
+### Tag: package-pure-2017-10-preview and java
+
+These settings apply only when `--tag=package-pure-2017-10-preview --java` is specified on the command line.
+Please also specify `--azure-libraries-for-java=<path to the root directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(tag) == 'package-pure-2017-10-preview' && $(java) && $(multiapi)
+java:
+  namespace: com.microsoft.azure.management.sql.v2017_10_01_preview
+  output-folder: $(azure-libraries-for-java-folder)/sql/resource-manager/v2017_10_01_preview
+regenerate-manager: true
+generate-interface: true
+```
+
+### Tag: package-pure-2018-06-preview and java
+
+These settings apply only when `--tag=package-pure-2018-06-preview --java` is specified on the command line.
+Please also specify `--azure-libraries-for-java=<path to the root directory of your azure-sdk-for-java clone>`.
+
+``` yaml $(tag) == 'package-pure-2018-06-preview' && $(java) && $(multiapi)
+java:
+  namespace: com.microsoft.azure.management.sql.v2018_06_01_preview
+  output-folder: $(azure-libraries-for-java-folder)/sql/resource-manager/v2018_06_01_preview
+regenerate-manager: true
+generate-interface: true
+```
+
+### Tag: package-pure-fixed-2017-03-preview
+
+These settings apply only when `--tag=package-pure-fixed-2017-03-preview` is specified on the command line.
+
+This section contains all input swagger files for version 2017-03-01-preview. All APIs of that version must be added this section when the API is ready for production.
+
+APIs must only be added to this section when the API is publicly available in at least 1 production region and at least 1 generated client has been tested end-to-end.
+
+These can be regenerated by running the following PowerShell script from this readme file's folder: `dir .\Microsoft.Sql\preview\2017-03-01-preview\ -File | Resolve-Path -Relative | % { " - $_".Replace("\", "/") }`
```

Comment:
Please explain in tag description how it is different from other tags :) What was the problem with `dataWarehouseUserActivities.json` ?

Reply:
@jaredmoo investigating :) for some reason it is hitting autorest exception - https://github.com/Azure/azure-rest-api-specs/runs/40229948 . I'm trying to isolate the error. Error message does not make sense :(

Reply:
@jaredmoo ok I found the issue. The problem is that in `2017-03-01-preview\dataWarehouseUserActivities.json` operationIds are named as `dataWarehouseUserActivities` and there is actually model definition which contain model with the name `dataWarehouseUserActivities` and that creates generation collision. Fortunately for java multi-api generation I can add a directive to rename `dataWarehouseUserActivities` type to `dataWarehouseUserActivity` and get everything generated. 

@jaredmoo I think that model should be renamed inside swagger. Why is it in plural form? it does not seems to contain any array or lists of objects.

Reply:
Ok, adding directive sounds awesome so that we don't need to add a new tag.

The reason it's plural is that it represents the aggregate status of recent activities, not just a single activity per resource. This collision is unfortunate, we could have renamed it on initial checkin if we knew about this problem :( but I want to maintain backcompat.